pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -253,13 +253,14 @@ func (srv *ProvisioningSrv) RouteRouteGetAlertRule(c *models.ReqContext, UID str
 	return response.JSON(http.StatusOK, definitions.NewAlertRule(rule, provenace))
 }
 
-func (srv *ProvisioningSrv) RoutePostAlertRule(c *models.ReqContext, ar definitions.ProvisionedAlertRule) response.Response {
+func (srv *ProvisioningSrv) RoutePostAlertRule(c *models.ReqContext, ar definitions.ProvisionedAlertRule, dp bool) response.Response {
```

Comment:
What is `dp`?

```suggestion
func (srv *ProvisioningSrv) RoutePostAlertRule(c *models.ReqContext, ar definitions.ProvisionedAlertRule, dp bool) response.Response {
```

Reply:
I'll change the naming

Reply:
Also, think about how the code flows in general; boolean parameters are an anti-pattern and try to come up with something different. (Rel: https://martinfowler.com/bliki/FlagArgument.html)

Reply:
You could i.e. directly pass the provenance instead of a boolean flag.